### PR TITLE
[gecko] Various improvements to the Firefox docker image

### PR DESCRIPTION
**Pull request by sylvestre**

### CODE REVIEW

**Comment by sylvestre** [comment]
@jankeromnes works for you?

**Review by jankeromnes — COMMENTED** [review]
Many thanks for this Pull Request!

I've just triggered a CI build, to test that the image build works (and then we'll also be able to test the resulting Docker image with Gitpod):

Screenshot 2020-10-21 at 09 01 21

I've also added a few thoughts/suggestions in-line.

I'll report back when the CI succeeds or fails.

File: gecko/Dockerfile [hunk 1]
```diff
@@ -25,20 +34,10 @@ RUN git clone https://github.com/glandium/git-cinnabar $HOME/.git-cinnabar \
...
- && git clone https://github.com/phacility/arcanist \
- && echo "\n# Phabricator helper." >> $HOME/.bashrc \
- && echo "PATH=\"\$PATH:$HOME/.phacility/arcanist/bin\"" >> $HOME/.bashrc
-
-# Install Phlay to support uploading multiple commits to Phabricator.
-RUN git clone https://github.com/mystor/phlay/ $HOME/.phlay \
- && echo "\n# Add Phlay to the PATH." >> $HOME/.bashrc \
- && echo "PATH=\"\$PATH:$HOME/.phlay\"" >> $HOME/.bashrc
+# Inject the path update to the top as bash completion needs to call rustc
+RUN sed -i '1s|^|PATH=\"\$PATH:$HOME/.cargo/bin/\"\n|' $HOME/.bashrc
```
[the remark this answers is not in the document]
Comment (sylvestre) [reply gecko/Dockerfile:38]: Are you sure? When I tried, I didn't have rustc in the path

File: gecko/Dockerfile [hunk 2]
```diff
@@ -7,16 +7,25 @@ RUN __RR_VERSION__="5.3.0" \
...
- && sudo rm -rf /tmp/gecko /var/lib/apt/lists/*
+ && sudo rm -rf /tmp/gecko /var/lib/apt/lists/* \
+
+# Install moz-phab
+ENV SHELL=/bin/bash
+RUN sudo apt-get update \
+ && sudo apt install -y python3-pip \
+ && pip3 install MozPhab \
+ && echo "PATH=\"\$PATH:$HOME/.local/bin/moz-phab \"" >> $HOME/.bashrc \
+ && sudo rm -rf /var/lib/apt/lists/*
```
[the remark this answers is not in the document]
Comment (sylvestre) [reply gecko/Dockerfile]: remove mach install-moz-phab instead (it wasn't doing the job when I tried)

**Comment by sylvestre** [comment]
@jankeromnes ping? :)

File: gecko/Dockerfile [hunk 3]
```diff
@@ -7,38 +7,38 @@ RUN __RR_VERSION__="5.3.0" \
...
  && cd /tmp/gecko \
  && ./mach bootstrap --no-interactive --application-choice=browser \
  && sudo apt-get remove -y nodejs nodejs-doc \
  && sudo rm -rf /tmp/gecko /var/lib/apt/lists/*
 
+# Install moz-phab
+ENV SHELL=/bin/bash
+RUN sudo apt-get update \
+ && pip3 install MozPhab \
+ && echo "PATH=\"\$PATH:$HOME/.local/bin/moz-phab \"" >> $HOME/.bashrc \
```
Comment (jankeromnes) [inline gecko/Dockerfile:25]: I'm not sure the space is useful here:

```suggestion
 && echo "PATH=\"\$PATH:$HOME/.local/bin/moz-phab\"" >> $HOME/.bashrc \
```

**Review by jankeromnes — APPROVED** [review]
Thanks again for this Pull Request! And sorry for the long response time, as I was on holidays.

I have good news: The build of `gitpod/workspace-gecko:branch-pr287` was successful! 🎉 And this image can be tested with:

https://gitpod.io/#https://github.com/jankeromnes/gecko-test

(without a Gecko clone yet, as that takes ages to checkout -- this is just to confirm that the image basically works, next up I'll create a reproducible Gitpod Snapshot of a fully-compiled Firefox checkout)

FYI, this is what the resulting `$PATH` looks like in Gitpod:

> /home/gitpod/.cargo/bin:/home/gitpod/.pyenv/plugins/pyenv-virtualenv/shims:/home/gitpod/.pyenv/shims:/workspace/.cargo/bin:/workspace/.pip-modules/bin:/workspace/.rvm/bin:/home/gitpod/.cargo/bin:/home/gitpod/.sdkman/candidates/maven/current/bin:/home/gitpod/.sdkman/candidates/java/current/bin:/home/gitpod/.sdkman/candidates/gradle/current/bin:/home/gitpod/.cargo/bin:/home/gitpod/.rvm/gems/ruby-2.6.6/bin:/home/gitpod/.rvm/gems/ruby-2.6.6@global/bin:/home/gitpod/.rvm/rubies/ruby-2.6.6/bin:/home/gitpod/.pyenv/plugins/pyenv-virtualenv/shims:/home/gitpod/.pyenv/shims:/workspace/.cargo/bin:/workspace/.pip-modules/bin:/workspace/.rvm/bin:/home/gitpod/.pyenv/bin:/home/gitpod/.pyenv/shims:/workspace/go/bin:/home/gitpod/go/bin:/home/gitpod/go-packages/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/home/linuxbrew/.linuxbrew/bin:/home/linuxbrew/.linuxbrew/sbin/:/home/gitpod/.nvm/versions/node/v12.18.4/bin:/home/gitpod/.git-cinnabar:/home/gitpod/.cargo/bin/:/home/gitpod/.rvm/bin:/home/gitpod/.rvm/bin:/home/gitpod/.local/bin/moz-phab :/home/gitpod/.git-cinnabar:/home/gitpod/.cargo/bin/:/home/gitpod/.rvm/bin:/home/gitpod/.local/bin/moz-phab :/home/gitpod/.git-cinnabar

For some reason, `moz-phab` appears 2 times, `git-cinnabar` 3 times and `.cargo/bin` 7 times (😬) but that should still work fine. Maybe let's just remove the space after `moz-phab` and call it done. 🚀

File: gecko/Dockerfile (hunk 3 above)
Comment (jankeromnes) [reply gecko/Dockerfile:25]: Ah wait, it seems that `/home/gitpod/.local/bin` doesn't even exist.

But in Gitpod, I can already find `moz-phab` here:

```
$ which moz-phab
/home/gitpod/.pyenv/shims/moz-phab
```

So maybe let's just remove this line:

```suggestion
```

**Comment by sylvestre** [comment]
Merci :+1:

**Comment by sylvestre** [comment]
Merci <3

**Pull request merged**